scripts/train/train.py

```diff
@@ -164,170 +164,264 @@ def build_dataloader(cfg: DictConfig, tokenizer: PreTrainedTokenizerBase,
+    device_eval_batch_size: int = cfg.pop('device_eval_batch_size')
+    max_duration: Union[int, str] = cfg.pop('max_duration')
+    eval_interval: Union[int, str] = cfg.pop('eval_interval')
+    precision: str = cfg.pop('precision')
+    max_seq_len: int = cfg.pop('max_seq_len')
+
+    # Optional parameters for will be set to default values if not specified.
+    default_run_name: str = os.environ.get('RUN_NAME', 'llm')
+    run_name: str = cfg.pop('run_name', default_run_name)
+    save_folder: Optional[str] = cfg.pop('save_folder', None)
+    save_latest_filename: str = cfg.pop('save_latest_filename',
+                                        'latest-rank{rank}.pt')
+    save_overwrite: bool = cfg.pop('save_overwrite', False)
+    save_weights_only: bool = cfg.pop('save_weights_only', False)
+    save_filename: str = cfg.pop('save_filename',
+                                 'ep{epoch}-ba{batch}-rank{rank}.pt')
+    save_interval: Union[str, int] = cfg.pop('save_interval', '1000ba')
+    save_num_checkpoints_to_keep: int = cfg.pop('save_num_checkpoints_to_keep',
+                                                -1)
+    progress_bar = cfg.pop('progress_bar', False)
+    log_to_console: bool = cfg.pop('log_to_console', True)
+    python_log_level: str = cfg.pop('python_log_level', 'debug')
+    console_log_interval: Union[int, str] = cfg.pop('console_log_interval',
+                                                    '1ba')
+    device_train_microbatch_size: Union[str, int] = cfg.pop(
+        'device_train_microbatch_size', 'auto')
+    eval_subset_num_batches: int = cfg.pop('eval_subset_num_batches', -1)
+    eval_first: bool = cfg.pop('eval_first', False)
+    load_path: str = cfg.pop('load_path', None)
+    load_weights_only: bool = cfg.pop('load_weights_only', False)
+    load_ignore_keys: Optional[List[str]] = cfg.pop('load_ignore_keys', None)
+    # Enable autoresume from model checkpoints if possible
+    autoresume_default = not save_overwrite and not save_weights_only
+    autoresume = cfg.pop('autoresume', autoresume_default)
+
+    # Pop known unused parameters.
+    cfg.pop('data_local', None)
+    cfg.pop('data_remote', None)
+    cfg.pop('global_seed')
+    cfg.pop('global_train_batch_size')
+    cfg.pop('n_gpus')
+    cfg.pop('device_train_grad_accum')
+
+    # Warn users for unused parameters
+    for key in cfg:
+        warnings.warn(
+            f'Unused parameter {key} found in cfg. Please check your yaml to ensure this parameter is necessary.'
+        )
+
+    # Warn if fsdp is enabled but user only has 1 GPU
     if dist.get_world_size() == 1 and fsdp_config is not None:
         warnings.warn(
             'FSDP is not applicable for single-GPU training. Reverting to DDP.')
-        cfg.pop('fsdp_config')
         fsdp_config = None
 
-    init_context = process_init_device(cfg.model, fsdp_config)
+    # Initialize context
+    init_context = process_init_device(model_config, fsdp_config)
```

Comment:
This call modifies the config, so we should log the config after it

Reply:
logged